src/Timeout/Timeout.php

```diff
@@ -0,0 +1,92 @@
+<?php declare(strict_types=1);
+
+namespace Bref\Timeout;
+
+/**
+ * Helper class to trigger an exception just before the Lamba times out. This
+ * will give the application a chance to shut down.
+ */
+final class Timeout
+{
+    /** @var bool */
+    private static $initialized = false;
+
+    /**
+     * Read environment variables and setup timeout exception.
+     */
+    public static function enable(): void
+    {
+        if (isset($_SERVER['BREF_TIMEOUT'])) {
+            $timeout = (int) $_SERVER['BREF_TIMEOUT'];
+            if ($timeout === -1) {
+                return;
+            }
+
+            if ($timeout > 0) {
+                self::timeoutAfter($timeout);
+
+                return;
+            }
+
+            // else if 0, continue
+        }
+
+        if (isset($_SERVER['LAMBDA_INVOCATION_CONTEXT'])) {
+            $context = json_decode($_SERVER['LAMBDA_INVOCATION_CONTEXT'], true, 512, JSON_THROW_ON_ERROR);
+            $deadlineMs = $context['deadlineMs'];
+            $remainingTime = $deadlineMs - intval(microtime(true) * 1000);
+
+            self::timeoutAfter((int) floor($remainingTime / 1000));
+
+            return;
+        }
+
+        throw new \LogicException('Could not find value for bref timeout. Are we running on Lambda?');
+    }
+
+    /**
+     * Setup custom handler for SIGTERM. One need to call Timeout::timoutAfter()
+     * to make an exception to be thrown.
+     *
+     * @return bool true if successful.
+     */
+    public static function init(): bool
```

Comment:
`init()` is called already by `timeoutAfter()`, how about making it private?

That would simplify the API of the `Timeout` class, and simplify the code in `LambdaRuntime`.

Reply:
But then the LambdaRuntime would not know if the timeout failed to setup. Currently `LambdaRuntime` disables the "exception throwing" if init fails. Ie, it will not try to init over and over again. 

https://github.com/brefphp/bref/pull/895/files#diff-042bba61d17e1d04b45e10c2440aab63bc6e32d5ca42c57fa67d5e9d56512f2cR69-R72

Making `init()` private would fill up the logs with potential error messages. 

....
Though.. one could argue that pcntl is always available and you would "only" get one error message per request.

Reply:
I did remove it.. the patch did not have much effect on `LambdaRuntime` 

https://github.com/brefphp/bref/pull/895/commits/52f8d65f665f7f0be6ee5b0de13a70694466c1e4